### PR TITLE
[processor] Implement WPTReport and tests

### PR DESCRIPTION
**Pull request by Hexcles**
WPTReport abstracts wptreport.json and provides some helper functions to
summarize and split the full report (the split output is needed by the
current frontend).

The code is largely based on
https://github.com/web-platform-tests/results-collection/blob/master/src/scripts/upload-wpt-results.py
but without any sharding support.

wptreport.py is also an executable (requires Python 3), a handy script
to summarize and/or split a wptreport JSON locally. See its help message
for detailed usage.

The plan is to run unit tests via `tox`, which takes care of creating
virtualenv and installing dependencies. The setup currently works
locally as long as Python 3 and tox are available. It hasn't been
integrated into Travis or the project Makefile yet; the integration will
happen soon later.

### CODE REVIEW

**Comment by Hexcles** [comment]
@lukebjerring FYI, this achieves the same purpose as your https://github.com/web-platform-tests/results-collection/pull/551

As I mentioned to you earlier, since this class is needed by my processor anyway, it probably makes more sense for me to make the class an executable script instead of adding another script to `results-collection`. Here's a more elaborated rationale: our use cases are exactly the same, whereas in `results-collection` there are some extra complexities because of the sharding support and BuildBot integration, so we can have a more concise, and perhaps better tested/exercised code path here, and in the meantime give `results-collection` more flexibility in case @jugglinmike wants to change the upload script.

Run `results-processor/wptreport.py -h` to try it out.

File: results-processor/tox.ini [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[tox]
+envlist = py3
+# We don't have or need setup.py for now.
+skipsdist=True
+
+[testenv]
+deps = -rrequirements.txt
```
Comment (lukebjerring) [inline results-processor/tox.ini]: This look suspiciously like a typo?

Comment (Hexcles) [reply results-processor/tox.ini]: Actually, no, though I agree it's weird... It's in fact equivalent to `-r requirements.txt` in the command line but somehow `tox` doesn't like the space (the syntax is documented in one of its basic examples).

**Comment by Hexcles** [comment]
Note: this is part of #146
